### DISCUSSION TITLE
Events not firing for javascript dynamically created events

### DISCUSSION THREAD

**Discussion by lambaek** [plausible/analytics#2133] [Anything Else]
Hi,

I'm working on a site, where i'm swapping elements dynamically based on what the user click.

I've run into an issue, where Plausible doesn't track when i click on one of these dynamic created elements, even though i have set up the data-analytics attribute the same as everywhere else on site

`data-analytics="&quot;Click affiliate&quot;, {&quot;props&quot;:{&quot;layout&quot;:&quot;offer_interactive_1&quot;,&quot;element&quot;:&quot;Logo&quot;,&quot;provider&quot;:&quot;providerName&quot;,&quot;location&quot;:&quot;/free-spins&quot;}}"`

I've tried adding click events to my new dynamic created element, but it doesn't help Plausible pick up when a user clicks it.

What am i missing? I'm sure it's me not doing something super simple..

**Comment by RobertJoonas** [comment]
Hi @Iambaek!

Do you see some error messages in the browser console perhaps? Is it a link element that you want to track? [This code example](https://plausible.io/docs/custom-event-goals#2-add-the-javascript-that-will-be-sending-the-link-click-events-to-plausible) only works with link (`<a>`) elements.

If it's a link element, and no error messages in the console, then another possibility is that the JavaScript code is located above the link element which doesn't work. The JavaScript code should come just before the closing `</body>` tag.

Hope this helps. If not, feel free to [contact us](https://plausible.io/contact) with more details so that we can have a closer look, thanks!
